sphinx/builders/linkcheck.py

```diff
@@ -40,10 +40,11 @@
 
 uri_re = re.compile('([a-z]+:)?//')  # matches to foo:// and // (a protocol relative URL)
 
-Hyperlink = NamedTuple('Hyperlink', (('next_check', float),
-                                     ('uri', Optional[str]),
+Hyperlink = NamedTuple('Hyperlink', (('uri', Optional[str]),
```

Comment:
Now I'm thinking to enhance this object more. It would be better to add `baseurl` and `anchor`. I hope they will make the latter step (ex. https://github.com/sphinx-doc/sphinx/pull/8830).

Reply:
Absolutely! I perfectly agree with you. Originally, I thought new Hyperlink object is a named-tuple like this:
```
Hyperlink = NamedTuple('Hyperlink', (('fulluri', Optional[str]),
                                     ('base_uri', Optional[str]),
                                     ('anchor', Optional[str]),
                                     ('docname', Optional[str]),
                                     ('lineno', Optional[int])))
```

And `CheckRequest` will be replaced with this:
```
CheckRequest = NamedTuple('CheckRequest', (('next_check', float),
                                           ('base_uri', Optional[str]),
                                           ('hyperlinks', Optional[List[Hyperlink]])))
```

Reply:
Great! :+1:
Since `base_uri` and `anchor` are derived from URI, perhaps we can get away without storing the `base_uri` and `anchor`. We’ll see what’s convenient when we implement it.